### ISSUE TITLE
'z' height above bar for 'f', 't'?

### ISSUE THREAD

**Issue by dtzWill** [agarick/agave#34]
Hi!

I'm not actually sure if this is a feature or a bug, but thought I'd ask :).

Image demonstrating:

![fizjt](https://user-images.githubusercontent.com/817330/96302214-0068c080-0fbe-11eb-9ba0-f3c8fcddec5b.png)

I put 'z' in the middle so it pops.

I was staring closely at the glyphs after updating to v30 and noticed this and was wondering if it's intentional,
and assuming it is :), out of curiosity why? (if you don't mind explaining, I'm just curious ^_^)

**Comment by blobject** [comment]
I also wonder about this x-height irregularity sometimes. Short answer is: the design was dictated by the simplistic geometric scheme, and things just sort of worked out this way.

**A bit more detail**

"z" (as well as every other ascii letter except "f", "g", "i", "j", "p", "q", "t", "y") has an x-height of exactly 1024 (modulo slight overshoots). This specific number is very important to the idea behind agave and "having the numbers line up" to specific ratios is one of the core design tenets. When it came to glyphs with descenders, however, such a large x-height simply did not make sense nor look good. So "g", "p", "q", and "y" were pinched up, out of sheer necessity. This is perhaps where the "quirky" and "retro-inspired" qualifiers might come in.

This leaves us with "f", "i", "j", and "t", whose story is similar - smaller x-heights were necessary for these glyphs to be "generally" okay to look at. If the bars of "f" and "t" were raised, these two glyphs would not look proportionate (I checked). For "i" and "j", raising the top endpoints will make their dots too small or the space between too narrow.

In summary, this x-height irregularity is somewhat intentional, though perhaps not desirable. Also, to be clear, this was not introduced in v30, but has been present from the very beginning. What v30 *did* introduce was the raised x-heights of "b", "d", and "h" to match that of the regular glyphs like "x" and "z".

**Your feedback**

If you don't mind sharing, did you have some thoughts on this, or what was your experience? Does it get in the way of reading the text or make things look too wonky? Personally, I have not been bothered by this (specifically by "f" and "t"; I was always a bit uneasy with "b" and "d" so I changed it in the latest version), but it would be nice to hear someone else's perspective. If the design is worth revisiting, I will do so and make changes.

**Comment by blobject** [comment]
Closing as poster doesn't seem to have more to add. Please reopen otherwise.
